test/contrib/sinatra/tracer_test.rb

```diff
@@ -237,4 +244,28 @@ def test_literal_template
     assert_equal(0, root.status)
     assert_nil(root.parent)
   end
+
+  def test_tagging_request_headers
```

Comment:
Is this test covering all the header options? I think it could use some more assertions around both request and response tags. I might suggest looking at [Rack's header tagging test](https://github.com/DataDog/dd-trace-rb/blob/0.13-dev/test/contrib/rack/middleware_test.rb#L365), which might not be perfect, but has some identical behavior its testing for.

Reply:
I've added tests for both cases. Which showed that Sinatra's `after` hook doesn't have all response headers yet, so I had to introduce simple rack Middleware to correctly handle that case.